ProcessQueue/__init__.py

```diff
@@ -159,320 +190,343 @@ def main(msg: func.QueueMessage):
-        table_ids.append(response[element]["id"])
-    # print("TABLE IDs", table_ids)
+    table_ids = [element["id"] for element in tables_content]
+
+    print("TABLE IDs", table_ids)
 
     """
     POST fields per table:
     For each row in Field Overview create an entry for scan_report_field,
     Empty row signifies end of fields in a table
-    Append field entry to data[] list,
+    Append field entry to field_entries_to_post[] list,
     Create JSON array with all the field entries, 
     Send POST request to API with JSON as input,
     Save the response data(field ids,field names) in a dictionary
     Set the current working sheet to be the same as the current table
     Post the values for that table
     """
-    idx = 0
-    data = []
+    table_idx = 0
+    field_entries_to_post = []
     names_x_ids = {}
+
+    # Loop over all rows in Field Overview sheet.
     # For sheets past the first two in the Scan Report
     # i.e. all 'data' sheets that are not Field Overview and Table Overview
     worksheet_idx = 2
+    print("Start fields loop", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
 
     for i, row_cell in enumerate(
-        ws.iter_rows(min_row=2, max_row=ws.max_row + 1), start=2
+        fo_ws.iter_rows(min_row=2, max_row=fo_ws.max_row + 1), start=2
     ):
 
-        if idx >= len(table_ids):
+        if table_idx >= len(table_ids):
             continue
         # If fraction empty or fraction unique is empty set to 0(decimal)
-        if not (ws.cell(row=i, column=8).value):
-            ws.cell(row=i, column=8).value = 0.0
-        if not (ws.cell(row=i, column=10).value):
-            ws.cell(row=i, column=10).value = 0.0
+        if not (fo_ws.cell(row=i, column=8).value):
+            fo_ws.cell(row=i, column=8).value = 0.0
+        if not (fo_ws.cell(row=i, column=10).value):
+            fo_ws.cell(row=i, column=10).value = 0.0
 
         # Create ScanReportField entry
-        scan_report_field_entry = {
-            "scan_report_table": table_ids[idx],
+        field_entry = {
+            "scan_report_table": table_ids[table_idx],
             "created_at": datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
             "updated_at": datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
-            "name": ws.cell(row=i, column=2).value,
-            "description_column": str(ws.cell(row=i, column=3).value),
-            "type_column": str(ws.cell(row=i, column=4).value),
-            "max_length": ws.cell(row=i, column=5).value,
-            "nrows": ws.cell(row=i, column=6).value,
-            "nrows_checked": ws.cell(row=i, column=7).value,
-            "fraction_empty": round(ws.cell(row=i, column=8).value, 2),
-            "nunique_values": ws.cell(row=i, column=9).value,
-            "fraction_unique": round(ws.cell(row=i, column=10).value, 2),
-            "flag_column": str(ws.cell(row=i, column=11).value),
+            "name": str(fo_ws.cell(row=i, column=2).value)[:63],

```

Comment:
We don't throw an error or anything if the name is too long, just truncate and will be fine? I guess this is just the name so should be (?)

Reply:
Good catch - this wasn't meant to go into production! Thanks.